src/server/server.cc

```diff
@@ -1215,24 +1215,25 @@ Server::InfoEntries Server::GetReplicationInfo() {
   }
 
   int idx = 0;
-  rocksdb::SequenceNumber latest_seq = storage->LatestSeqNumber();
 
   {
     std::shared_lock<std::shared_mutex> guard(slave_threads_mu_);
     entries.emplace_back("connected_slaves", slave_threads_.size());
+    rocksdb::SequenceNumber latest_seq = storage->LatestSeqNumber();
     for (const auto &slave : slave_threads_) {
       if (slave->IsStopped()) continue;
 
+      auto slave_ack_seq = slave->GetAckSeq();
       entries.emplace_back(
           "slave" + std::to_string(idx),
           fmt::format("ip={},port={},offset={},lag={}", slave->GetConn()->GetAnnounceIP(),
-                      slave->GetConn()->GetAnnouncePort(), slave->GetAckSeq(), latest_seq - slave->GetAckSeq()));
+                      slave->GetConn()->GetAnnouncePort(), slave_ack_seq >= latest_seq ? latest_seq : slave_ack_seq,
```

Comment:
```suggestion
                      slave->GetConn()->GetAnnouncePort(), slave_ack_seq,
```
slave_ack_seq should be enough here?

Reply:
if slave_ack_seq > latest_seq, then offset > master_repl_offset, but the lag is 0, it looks like mismatch

Reply:
Sounds reasonable.